### PR TITLE
Fix issue 3155 by running `scancode-reindex-licenses` subcommand instead of using `--reindex-licenses` flag

### LINKED ISSUE #3155: Dockerfile uses removed --reindex-licenses flag
### Description

The Dockerfile uses the `--reindex-licenses` flag 
https://github.com/nexB/scancode-toolkit/blob/develop/Dockerfile#L42
which was recently removed
https://github.com/nexB/scancode-toolkit/commit/5361052117189489703d436a68e5f364f9a39c0d#diff-fad3fa964c566642a9a64df45efd8c9daeb435e1c8791ac1f64d2881b299d753L165
in this PR
https://github.com/nexB/scancode-toolkit/pull/2979

so the Docker image no longer builds.

### How To Reproduce

Follow install instructions on 
https://scancode-toolkit.readthedocs.io/en/latest/getting-started/install.html#installation-via-docker

```
git clone https://github.com/nexB/scancode-toolkit
cd scancode-toolkit
docker build --tag scancode-toolkit --tag scancode-toolkit:$(git describe --tags) .
```

Result
```
 => ERROR [5/5] RUN ./scancode --reindex-licenses                        134.4s
------
 > [5/5] RUN ./scancode --reindex-licenses:
#9 0.370 * Configuring ScanCode for first use...
#9 133.8 Usage: scancode [OPTIONS] <OUTPUT FORMAT OPTION(s)> <input>...
#9 133.8 Try the 'scancode --help' option for help on options and arguments.
#9 133.8
#9 133.8 Error: No such option: --reindex-licenses (Possible options: --is-license-text, --license, --unknown-licenses)
------
executor failed running [/bin/sh -c ./scancode --reindex-licenses]: exit code: 2
```

### System configuration

MacOS 13.0.1

### PR DESCRIPTION
**Pull request by abhi-kr-2100**
Fixes #3155 

This PR only modifies the Dockerfile. In the erroneous Dockerfile, `./scancode --reindex-licenses` was used to
1. Configure venv
2. Reindex licenses

Since the `--reindex-licenses` flag has been removed, this led to an error. The fix is to use the new subcommand `scancode-reindex-licenses` to do the reindexing. Configuration is done prior to the reindexing using the `./configure` script.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

Signed-off-by: Abhishek Kumar <abhi.kr.2100@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -37,9 +37,11 @@ WORKDIR /scancode-toolkit
...
 COPY . /scancode-toolkit
 
-# Run scancode once for initial configuration, with 
-# --reindex-licenses to create the base license index
-RUN ./scancode --reindex-licenses
+# Initial configuration
+RUN ./configure
+
+# Run scancode-reindex-licenses to build the base license index
+RUN ./venv/bin/scancode-reindex-licenses
```
Comment (s01ipsist) [inline Dockerfile]: Dockerfile best practice is to combine successive `RUN` commands so as to minimise image layers.

```
RUN ./configure && \
    ./venv/bin/scancode-reindex-licenses
```
Use comments as you prefer

File: azure-pipelines.yml [hunk 2]
```diff
@@ -7,6 +7,27 @@
...
+    - task: Docker@2
+      displayName: Check that building using Docker succeeds
+      inputs:
+        command: 'build'
+        Dockerfile: '**/Dockerfile'
+        tags: |
+          'scancode-toolkit'
+          'scancode-toolkit:$(Build.BuildId)'
+    - script: |
+        docker scan scancode-toolkit:$(Build.BuildId)
```
Comment (s01ipsist) [inline azure-pipelines.yml]: I understand the recommendation was to use the built image to run a scan.
https://scancode-toolkit.readthedocs.io/en/latest/getting-started/install.html#run-using-docker
Has some sample commands

File: Dockerfile (hunk 1 above)
Comment (abhi-kr-2100) [reply Dockerfile]: Thank for taking the time to review.

I've combined the two into a single RUN instruction.

File: azure-pipelines.yml (hunk 2 above)
Comment (abhi-kr-2100) [reply azure-pipelines.yml]: Thanks a lot for pointing me in the right direction; I had gotten myself on a completely wrong track.

I've updated the `script` key with a sample run of `scancode`.

**Review by s01ipsist — COMMENTED** [review]
Looks ok to me @abhi-kr-2100 but we need a project maintainer to review

**Comment by abhi-kr-2100** [comment]
Thanks for taking the time to review this @s01ipsist.

Let me gently ping @pombredanne.

**Review by pombredanne — APPROVED** [review]
LGTM! thank you ++
Merging now

**Comment by pombredanne** [comment]
@abhi-kr-2100 Thanks!

**Comment by pombredanne** [comment]
@abhi-kr-2100 I did not pay attention to the details, but your changes have disabled the Azure pipelines entirely. I am going to revert this urgently

**Comment by abhi-kr-2100** [comment]
@pombredanne Oops. Please do the needful.

**Pull request merged**